app/ide-desktop/lib/dashboard/src/authentication/src/hooks.tsx

```diff
@@ -0,0 +1,53 @@
+/** @file Module containing common custom React hooks used throughout out Dashboard. */
+import * as react from "react";
+import * as loggerProvider from "./providers/logger";
+
+// ======================
+// === useAsyncEffect ===
+// ======================
+
+/** A React hook for re-rendering a component once an asynchronous call is over.
+ *
+ * This hook will take care of setting an initial value for the component state (so that it can
+ * render immediately), updating the state once the asynchronous call is over (to re-render the
+ * component), and cancelling any in-progress asynchronous calls when the component is unmounted (to
+ * avoid race conditions where "update 1" starts, "update 2" starts and finishes, then "update 1"
+ * finishes and sets the state).
+ *
+ * For further details, see: https://devtrium.com/posts/async-functions-useeffect
+ * Also see: https://stackoverflow.com/questions/61751728/asynchronous-calls-with-react-usememo
+ *
+ * @param initialValue - The initial value of the state controlled by this hook.
+ * @param fetch - The asynchronous function used to load the state controlled by this hook.
+ * @param deps - The list of dependencies that, when updated, trigger the asynchronous fetch.
+ * @returns value - The current value of the state controlled by this hook. */
+export function useAsyncEffect<T>(
+  initialValue: T,
+  fetch: () => Promise<T>,
+  deps?: react.DependencyList
+): T {
+  const logger = loggerProvider.useLogger();
+  const [value, setValue] = react.useState<T>(initialValue);
+
+  react.useEffect(() => {
+    let active = true;
+
+    /** Declare the async data fetching function. */
+    const load = async () => {
+      const result = await fetch();
+
+      /** Set state with the result if `active` is true. This prevents race conditions by
+       * making it so that only the latest async fetch will update the state on completion. */
+      if (!active) return;
+      setValue(result);
+    };
+
+    load().catch((error) => logger.error("Error while fetching data", error));
+    /** Cancel any future `setValue` calls. */
+    return () => {
+      active = false;
```

Comment:
consider using [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) which should automatically abort `fetch`es. MDN says it's supported by node 15+ (and chrome 66+)

as a bonus, it should make `load` simpler if not redundant - something like this:
```ts
const abort = new AbortController();
fetch({ signal: abort.signal }).then(setValue).catch((error) => logger.error("Error while fetching data", error));
```
it would require the signature of `fetch` to be changed, but consumers with 0-argument fetch will still work, since they can ignore the argument

how useful this change would be would probably depend on how many consumers are using the web API `fetch`